core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/edit-ema-editor.component.ts

```diff
@@ -433,15 +421,15 @@ export class EditEmaEditorComponent implements OnInit, OnDestroy {
      * @memberof EditEmaEditorComponent
      */
     updateCurrentDevice(device: DotDevice & { icon?: string }) {
-        this.store.updatePreviewState({
-            editorMode: EDITOR_MODE.PREVIEW,
+        this.store.updateEditorData({
+            mode: EDITOR_MODE.DEVICE,
             device
         });
```

Comment:
```suggestion
        this.store.setDevice(device);
```

Or something like that.